snapcraft/cli/templates.py

```diff
@@ -0,0 +1,68 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import os
+import yaml
+import sys
+
+from ._options import get_project
+from snapcraft.internal import common, project_loader
+
+
+@click.group()
+def templatecli(**kwargs):
+    pass
+
+
+@templatecli.command()
+def templates(**kwargs):
+    """List available templates."""
+
+    template_names = os.listdir(common.get_templatesdir())
+    if not template_names:
+        click.echo("No templates supported")
```

Comment:
maybe even tie it to the base? So then different people running it don't get confused with, hey I get some, what about you? Or is this going to list everything regardless of base?